lib/slice.js

```diff
@@ -0,0 +1,43 @@
+const isNumber = require( 'lodash/isNumber' );
+const inRange = require( 'lodash/inRange' );
+const assert = require( 'assert' );
+
+function createSliceStream( start, end = Infinity ){
+
+  assert( isNumber( start ) , 'start must be a number' );
+  assert( start >= 0, 'start must be >= 0' );
+  assert( isNumber( end ), 'end must be a number' );
+
+  let index = 0;
+
+  function transform( item, _, next ){
+
+    if( index == end ){
+
+      this.unpipe();
```

Comment:
you need to unpipe from the streams that piped in - this here unpipes from follow streams that this stream piped into. I've sent you a work example...